intel-retail/automated-self-checkout
# feat: enhance the ovms client infra-structure with the capability to configure env files from /envs directory

## PR Checklist



## What are you changing?


## Issue this PR will close

close: #**issue_number**

## Anything the reviewer should know when reviewing this PR?

## Test Instructions if applicable

- unit test: make unit-test-ovms-client
- integration tests:  
1. rebuild ovms-client docker images: `make build-ovms-client` 
2. edit any configuration.yml file of any ovms profile, for example object_detection to add the following yaml snippet into configuration.yml for environment variable files to be applied to that profile:
```yml
    EnvironmentVariableFiles:
      - test1.env
      - test2.env
```
3. rebuild that docker image of object_detection: eg: `make build-python-apps`
4. start the camera-simulator: ./camera-simulator/camera-simulator.sh
5. run it with docker-run.sh : `PIPELINE_PROFILE="object_detection" sudo -E ./docker-run.sh --pipelines 1 --logdir mytest/data --duration 120 --init_duration 60 --platform core --inputsrc rtsp://127.0.0.1:8554/camera_0 --workload opencv-ovms`
6. do a docker logs ovms-client0 and you should see that the environment variables that are in the env files (test1.env and test2.env) now can be found into docker logs.

## If the there are associated PRs in other repositories, please link them here (i.e. intel-retail/automated-self-checkout )
